src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -0,0 +1,236 @@
+      this.debouncedInput = value;
+    }, 500),
+    fuzzyMatch({ debouncedInput, symbols, processedInputRegex }) {
+      const matches = [];
+      symbols.map((symbol) => {
+        if (debouncedInput) {
+          const match = processedInputRegex.exec(symbol.title.toLowerCase());
+          if (match) {
+            const isExactMatch = RegExp(
+              debouncedInput,
+            ).test(symbol.title.toLowerCase());
+            if (Math.abs(match[0].length - debouncedInput.length) > debouncedInput.length * 2) {
+              return false;
+            }
+            matches.push({
+              title: symbol.title,
+              path: symbol.path,
+              inputLengthDifference: Math.abs(
+                symbol.title.length - debouncedInput.length,
+              ),
+              matchLength: match[0].length,
+              matchLengthDifference: isExactMatch === true
+                ? 0
+                : Math.abs(match[0].length - debouncedInput.length),
+              start: match.index,
+              substring: match[0],
+            });
+            return true;
+          }
+        }
+        return false;
+      });
+      return matches;
+    },
+    orderSymbolsByPriority(matchingSymbols) {
+      return matchingSymbols.sort((a, b) => {
+        // Sort exact string match by symbol title length
+        if (a.matchLengthDifference === 0 && b.matchLengthDifference === 0) {
+          if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+          if (a.inputLengthDifference < b.inputLengthDifference) return -1;
+        }
+        // Exact string match have priority over fuzzy match
+        if (a.matchLengthDifference === 0 || b.matchLengthDifference === 0) {
+          if (a.matchLengthDifference > b.matchLengthDifference) return 1;
+          if (a.matchLengthDifference < b.matchLengthDifference) return -1;
+        }
+        // Shortests symbol match title have preference over larger titles
+        if (a.matchLengthDifference > b.matchLengthDifference) return 1;
+        if (a.matchLengthDifference < b.matchLengthDifference) return -1;
+        // Shortests symbol title have preference over larger titles
+        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
+        // Matches at the beginning of string have relevance over matches at the end
+        if (a.start > b.start) return 1;
+        if (a.start < b.start) return -1;
+        return 0;
+      });
+    },
+  },
+  watch: {
+    userInput: 'debounceInput',
+  },
+};
+</script>
+
+<style scoped>
+
+  .close-icon {
+    padding-bottom: 10px;
+  }
+  .container {
+      flex-direction: row;
+  }
+  .filter {
+    border-radius: 5px;
+    background-color: white;
+    border: solid rgb(212, 212, 212) 1px;
+    padding: 20px;
+    width: 80%;
+    margin-bottom: 20px;
+  }
+  .quick-navigation-modal {
+    background-color: rgb(255, 255, 255);
```

Comment:
I understand these aren't variables because you will be focusing on the UI later on

Reply:
yes the UI will change in an upcoming pr, but I changed the hardcoded color value to a variable in c2d8214 anyway to keep the standard